rlog_test.go

```diff
@@ -0,0 +1,241 @@
+	for scanner.Scan() {
+		line := scanner.Text()
+		if timeLayout != "" {
+			dateTime := line[:len(timeLayout)]
+			line = line[len(timeLayout)+1:]
+			_, err := time.Parse(timeLayout, dateTime)
+			if err != nil {
+				t.Fatal(err)
+				t.Fatalf("Incorrect date/time format.\nSHOULD: %s\nIS:     %s\n", timeLayout, dateTime)
+			}
+		}
+		if i >= len(checkLines) {
+			t.Fatal("Not enough lines provided in checkLines.")
+		}
+		if line != checkLines[i] {
+			t.Fatalf("Log line %d does not match check line.\nSHOULD: %s\nIS:     %s\n", i, checkLines[i], line)
+		}
+		i++
+	}
+	if len(checkLines) > i {
+		t.Fatalf("Only %d of %d checklines found in output file.", i, len(checkLines))
+	}
+	if i == 0 {
+		t.Fatal("No input scanned")
+	}
+}
+
+func TestLogLevels(t *testing.T) {
+	conf := setup()
+	defer cleanup()
+
+	conf.logLevel = "DEBUG"
+	Initialize(conf)
+
+	Debug("Test Debug")
+	Info("Test Info")
+	Warn("Test Warning")
+	Error("Test Error")
+	Critical("Test Critical")
+
+	checkLines := []string{
+		"DEBUG    : Test Debug",
+		"INFO     : Test Info",
+		"WARN     : Test Warning",
+		"ERROR    : Test Error",
+		"CRITICAL : Test Critical",
+	}
+	fileMatch(t, checkLines, "")
+}
+
+func TestLogLevelsLimited(t *testing.T) {
+	conf := setup()
+	defer cleanup()
+
+	conf.logLevel = "WARN"
+	conf.traceLevel = "3"
+	Initialize(conf)
+
+	Debug("Test Debug")
+	Info("Test Info")
+	Warn("Test Warning")
+	Error("Test Error")
+	Critical("Test Critical")
+	Trace(1, "Trace 1")
+	Trace(2, "Trace 2")
+	Trace(3, "Trace 3")
+	Trace(4, "Trace 4")
+	checkLines := []string{
+		"WARN     : Test Warning",
+		"ERROR    : Test Error",
+		"CRITICAL : Test Critical",
+		"TRACE(1) : Trace 1",
+		"TRACE(2) : Trace 2",
+		"TRACE(3) : Trace 3",
+	}
+	fileMatch(t, checkLines, "")
+}
+
+func TestLogFormatted(t *testing.T) {
+	conf := setup()
+	defer cleanup()
+
+	conf.logLevel = "DEBUG"
+	conf.traceLevel = "1"
+	Initialize(conf)
+
+	Debugf("Test Debug %d", 123)
+	Infof("Test Info %d", 123)
+	Warnf("Test Warning %d", 123)
+	Errorf("Test Error %d", 123)
+	Criticalf("Test Critical %d", 123)
+	Tracef(1, "Trace 1 %d", 123)
+	checkLines := []string{
+		"DEBUG    : Test Debug 123",
+		"INFO     : Test Info 123",
+		"WARN     : Test Warning 123",
+		"ERROR    : Test Error 123",
+		"CRITICAL : Test Critical 123",
+		"TRACE(1) : Trace 1 123",
+	}
+	fileMatch(t, checkLines, "")
+}
+
+func TestLogTimestamp(t *testing.T) {
+	conf := setup()
+	defer cleanup()
+
+	conf.logTimeFormat = "ANSIC"
+	conf.logTimeDate = true
+	Initialize(conf)
+
+	Info("Test Info")
+	checkLines := []string{
+		"INFO     : Test Info",
+	}
+	fileMatch(t, checkLines, time.ANSIC)
+}
+
+func TestLogCallerInfo(t *testing.T) {
+	// In this test we manually figure out the caller info, which we should
+	// display
+	conf := setup()
+	defer cleanup()
+
+	conf.showCallerInfo = true
+	Initialize(conf)
+
+	Info("Test Info")
+	pc, fullFilePath, line, _ := runtime.Caller(0)
+	line-- // The log was called in the line before, so... -1
```

Comment:
nitpick: comment above line--, not on same one, kinda confusing becuase next line is different comment.